models/user.go

```diff
@@ -35,5 +35,9 @@ func GetUserByUsername(username string) (User, error) {
 // PutUser updates the given user
 func PutUser(u *User) error {
 	err := db.Save(u).Error
+	if err != nil {
```

Comment:
As mentioned before, let's remove these checks that aren't necessary for these changes.